Source/CSharpEssentials.Tests/UseGetterOnlyAutoPropertyAnalyzerTests.cs

```diff
@@ -0,0 +1,201 @@
+﻿using CSharpEssentials.GetterOnlyAutoProperty;
+using Microsoft.CodeAnalysis.Diagnostics;
+using NUnit.Framework;
+
+namespace CSharpEssentials.Tests
+{
+    [TestFixture]
+    public class UseGetterOnlyAutoPropertyAnalyzerTests : AnalyzerTestFixture
```

Comment:
A couple of tests to consider adding:

1. The diagnostic exists when there are multiple constructors that write to the property.
2. The diagnostic exists when the property's written in a constructor and read from another method.
3. The diagnostic exists when the property and the constructor writing to it are in different parts of a partial class.